server/apidata/apidata.go

```diff
@@ -118,17 +111,19 @@ func (s *DataAPI) ReportEpoch(base, quote uint32, epochIdx uint64, stats *matche
 	}
 
 	// Add the candlestick.
-	s.cacheMtx.Lock()
-	mktCaches := s.marketCaches[mktName]
-	if mktCaches == nil {
-		s.cacheMtx.Unlock()
-		return nil, fmt.Errorf("unknown market %q", mktName)
-	}
-	epochDur := s.epochDurations[mktName]
-	startStamp := epochIdx * epochDur
-	endStamp := startStamp + epochDur
-	for _, cache := range mktCaches {
-		cache.Add(&candles.Candle{
+	addCandle := func() (change24 float64, vol24 uint64, err error) {
+		s.cacheMtx.Lock()
+		defer s.cacheMtx.Unlock()
+		mktCaches := s.marketCaches[mktName]
+		if mktCaches == nil {
+			return 0, 0, fmt.Errorf("unknown market %q", mktName)
+		}
+		epochDur := s.epochDurations[mktName]
+		startStamp := epochIdx * epochDur
+		endStamp := startStamp + epochDur
+		var cache5min *candles.Cache
+		const fiveMins = uint64(time.Minute * 5 / time.Millisecond)
```

Comment:
Not sure if this is clearer than `(5*time.Minute).Milliseconds()` or just `5 * 60 * 1000`, but ok